packages/aws-cdk-lib/custom-resources/test/custom-resource-config/custom-resource-config.test.ts

```diff
@@ -0,0 +1,230 @@
+      RetentionInDays: nonCustomResourceLogRetention,
+    });
+  });
+});
+
+describe('when a singleton-backed custom resource logRetention is specified', () => {
+  test('addLogRetentionLifetime overrides log retention', () => {
+    const customResourceLogRetention = logs.RetentionDays.TEN_YEARS;
+    const app = new cdk.App();
+    CustomResourceConfig.of(app).addLogRetentionLifetime(customResourceLogRetention);
+    const stack = new cdk.Stack(app);
+
+    const websiteBucket = new s3.Bucket(stack, 'WebsiteBucket', {});
+    new s3deploy.BucketDeployment(stack, 's3deployLogRetention', {
+      sources: [s3deploy.Source.jsonData('file.json', { a: 'b' })],
+      destinationBucket: websiteBucket,
+      logRetention: logs.RetentionDays.ONE_WEEK,
+    });
+    const template = Template.fromStack(stack);
+
+    template.hasResourceProperties('Custom::LogRetention', {
+      RetentionInDays: customResourceLogRetention,
+    });
+    template.resourceCountIs('AWS::Logs::LogGroup', 1);
+    template.hasResourceProperties('AWS::Logs::LogGroup', {
+      RetentionInDays: customResourceLogRetention,
+    });
+  });
+});
+
+describe('when a singleton-backed custom resource log group is specified', () => {
+  test('addLogRetentionLifetime modifies the retention period of a singleton-backed custom resource log group.', () => {
+    const customResourceLogRetention = logs.RetentionDays.TEN_YEARS;
+    const app = new cdk.App();
+    CustomResourceConfig.of(app).addLogRetentionLifetime(customResourceLogRetention);
+    const stack = new cdk.Stack(app);
+
+    const websiteBucket = new s3.Bucket(stack, 'WebsiteBucket', {});
+    new s3deploy.BucketDeployment(stack, 's3deployLogGroup', {
+      sources: [s3deploy.Source.jsonData('file.json', { a: 'b' })],
+      destinationBucket: websiteBucket,
+      logGroup: new logs.LogGroup(stack, 'LogGroup', {
+        retention: logs.RetentionDays.ONE_WEEK,
+      }),
+    });
+
+    const template = Template.fromStack(stack);
+
+    template.resourceCountIs('AWS::Logs::LogGroup', 1);
+    template.hasResourceProperties('AWS::Logs::LogGroup', {
+      RetentionInDays: customResourceLogRetention,
+    });
+  });
+
+  test('addLogRetentionLifetime only modifies the singleton-backed custom resource log group', () => {
+    const customResourceLogRetention = logs.RetentionDays.TEN_YEARS;
+    const app = new cdk.App();
+    CustomResourceConfig.of(app).addLogRetentionLifetime(customResourceLogRetention);
+    const stack = new cdk.Stack(app);
+
+    const websiteBucket = new s3.Bucket(stack, 'WebsiteBucket', {});
+    new s3deploy.BucketDeployment(stack, 's3deployLogGroup1', {
+      sources: [s3deploy.Source.jsonData('file.json', { a: 'b' })],
+      destinationBucket: websiteBucket,
+      logGroup: new logs.LogGroup(stack, 'LogGroup1', {
+        retention: logs.RetentionDays.ONE_WEEK,
+      }),
+    });
+    new s3deploy.BucketDeployment(stack, 's3deployLogGroup2', {
```

Comment:
`s3deployLogGroup2` is not a good name for this construct, because it isn't a log group. Can we call it `BucketDeployment2`?